rct229/rulesets/ashrae9012019/section19/section19rule10.py

```diff
@@ -0,0 +1,271 @@
+                baseline_system_type_compare(system_type, applicable_sys_type, False)
+                for system_type in baseline_system_types_dict_b.keys()
+                for applicable_sys_type in APPLICABLE_SYS_TYPES
+            ]
+        )
+
+    def create_data(self, context, data):
+        rmr_b = context.baseline
+        rmi_b = rmr_b["ruleset_model_instances"][0]
+
+        rmr_p = context.proposed
+        assert_(
+            rmr_p["ruleset_model_instances"],
+            "Please make sure `ruleset_model_instances` exists in the `proposed`.",
+        )
+        rmi_p = rmr_p["ruleset_model_instances"][0]
+
+        proposed_has_economizer = is_economizer_modeled_in_proposed(rmi_b, rmi_p)
+
+        hvac_system_exception_2_list = []
+        if rmi_b.get("refrigerated_cases") is not None and not proposed_has_economizer:
+            hvac_system_exception_2_list = [
+                hvac_id_b
+                for hvac_id_b in find_all(
+                    f'$.buildings[*].building_segments[*][?(@.lighting_building_area_type = "{LIGHTING_BUILDING_AREA.RETAIL}")].heating_ventilating_air_conditioning_systems[*].id',
+                    rmi_b,
+                )
+            ]
+
+        baseline_system_types_dict = get_baseline_system_types(rmi_b)
+
+        return {
+            "proposed_has_economizer": proposed_has_economizer,
+            "hvac_system_exception_2_list": hvac_system_exception_2_list,
+            "HVAC_systems_primarily_serving_comp_rooms_list": get_hvac_systems_primarily_serving_comp_room(
+                rmi_b
+            ),
+            "baseline_system_types_dict_b": {
+                system_type: system_list
+                for system_type, system_list in baseline_system_types_dict.items()
+                if system_type in APPLICABLE_SYS_TYPES and system_list
+            },
+        }
+
+    class HVACRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section19Rule10.HVACRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, True),
+                required_fields={
+                    "$": ["fan_system"],
+                },
+            )
+
+        def is_applicable(self, context, data=None):
+            hvac_b = context.baseline
+            hvac_id_b = hvac_b["id"]
+            baseline_system_types_dict_b = data["baseline_system_types_dict_b"]
+
+            return any(
+                hvac_id_b in baseline_system_types_dict_b[system_type]
+                for system_type in baseline_system_types_dict_b.keys()
+            )
+
+        def get_calc_vals(self, context, data=None):
+            hvac_b = context.baseline
+            hvac_id_b = hvac_b["id"]
+
+            HVAC_systems_primarily_serving_comp_rooms_list = data[
+                "HVAC_systems_primarily_serving_comp_rooms_list"
+            ]
+            baseline_system_types_b = list(data["baseline_system_types_dict_b"].keys())[
+                0
+            ]
```

Comment:
The name of this parameter is confusing - suggest to revise to:
`baseline_system_type`

Also, I think the goal is to find the baseline system type by hvac_id.
So the code written in the RDS is correct
`system_type_b= list(baseline_system_types_dict.keys())[list(baseline_system_types_dict.values()).index(hvac_id_b)])`